modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -540,7 +551,12 @@ INSFVRhieChowInterpolator::getVelocity(const Moose::FV::InterpMethod m,
         &fi, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, boundary_elem};
     auto velocity = vel(boundary_face, time);
     incorporate_mesh_velocity(boundary_face, velocity);
-    return velocity;
+
+    // If not solving for velocity, clear derivatives
+    if (_fe_problem.currentNlSysNum() != _u->sys().number())
+      return {velocity(0).value(), velocity(1).value(), velocity(2).value()};
```

Comment:
I usually just use `raw_value` I think that would make it a little nicer. Same for the other changes.

Reply:
yeah I have no preference

Reply:
changed